From a4707faf5a38f9de12c111429dbdf1b98969dec3 Mon Sep 17 00:00:00 2001
From: Li Lingfeng <lilingfeng3@huawei.com>
Date: Sat, 13 Aug 2022 16:25:31 +0800
Subject: [PATCH 4573/4585] [Huawei] io_uring: add switch to control io_uring
 whether to enable

Offering: HULK
hulk inclusion
category: feature
bugzilla: 187472

---------------------------

Io_uring is disabled as default.
Inserting "iouring_enable" into cmdline to enable it.

Signed-off-by: Li Lingfeng <lilingfeng3@huawei.com>
---
 fs/io_uring.c | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/fs/io_uring.c b/fs/io_uring.c
index 257e4af176e6..cc9c72d050d7 100644
--- a/fs/io_uring.c
+++ b/fs/io_uring.c
@@ -986,6 +986,17 @@ static struct kmem_cache *req_cachep;
 
 static const struct file_operations io_uring_fops;
 
+
+static int io_uring_enable;
+
+static int __init io_uring_enable_setup(char *str)
+{
+	io_uring_enable = 1;
+	return 1;
+}
+
+__setup("iouring_enable", io_uring_enable_setup);
+
 struct sock *io_uring_get_socket(struct file *file)
 {
 #if defined(CONFIG_UNIX)
@@ -9094,6 +9105,9 @@ SYSCALL_DEFINE6(io_uring_enter, unsigned int, fd, u32, to_submit,
 	int submitted = 0;
 	struct fd f;
 
+	if (io_uring_enable == 0)
+		return -EOPNOTSUPP;
+
 	io_run_task_work();
 
 	if (flags & ~(IORING_ENTER_GETEVENTS | IORING_ENTER_SQ_WAKEUP |
@@ -9593,6 +9607,9 @@ static long io_uring_setup(u32 entries, struct io_uring_params __user *params)
 SYSCALL_DEFINE2(io_uring_setup, u32, entries,
 		struct io_uring_params __user *, params)
 {
+	if (io_uring_enable == 0)
+		return -EOPNOTSUPP;
+
 	return io_uring_setup(entries, params);
 }
 
@@ -9898,6 +9915,9 @@ SYSCALL_DEFINE4(io_uring_register, unsigned int, fd, unsigned int, opcode,
 	long ret = -EBADF;
 	struct fd f;
 
+	if (io_uring_enable == 0)
+		return -EOPNOTSUPP;
+
 	f = fdget(fd);
 	if (!f.file)
 		return -EBADF;
-- 
2.17.1

